src/transformers/pipelines.py

```diff
@@ -1180,19 +1185,60 @@ def __call__(
 
         """
         assert return_tensors or return_text, "You must specify return_tensors=True or return_text=True"
-        if self.framework == "tf":
-            raise NotImplementedError("Tensorflow not supported")
+        is_t5 = "T5ForConditionalGeneration" in self.model.__class__.__name__
+        is_bart = "BartForConditionalGeneration" in self.model.__class__.__name__
+
+        if self.framework == "tf" and is_bart:
+            raise NotImplementedError(
+                "Tensorflow is not yet supported for Bart. Please consider using T5, e.g. `t5-base`"
+            )
+
+        if is_t5:
+            documents = tuple(["summarize: " + document for document in list(documents)])
+
         with self.device_placement():
             inputs = self._parse_and_tokenize(*documents)
-            inputs = self.ensure_tensor_on_device(**inputs)
+
+            if self.framework == "pt":
+                inputs = self.ensure_tensor_on_device(**inputs)
+                input_length = inputs["input_ids"].shape[-1]
+            elif self.framework == "tf":
+                input_length = tf.shape(inputs["input_ids"])[-1]
+
+            if is_bart:
+                decoder_start_token_id = self.tokenizer.eos_token_id
+            elif is_t5:
+                decoder_start_token_id = self.tokenizer.pad_token_id
+            else:
+                decoder_start_token_id = None
+
+            if input_length < min_length // 2:
+                logger.warning(
+                    "Your min_length is set to {}, but you input_length is only {}. You might consider decreasing min_length, e.g. summarizer('...', min_length=...)".format(
+                        min_length, input_length
+                    )
+                )
+
+            if input_length < max_length:
+                logger.warning(
+                    "Your max_length is set to {}, but you input_length is only {}. You might consider decreasing max_length, e.g. summarizer('...',max_length=...)".format(
+                        max_length, input_length
+                    )
+                )
+
             summaries = self.model.generate(
                 inputs["input_ids"],
                 attention_mask=inputs["attention_mask"],
                 max_length=max_length,
                 min_length=min_length,
-                do_sample=False,
+                decoder_start_token_id=decoder_start_token_id,
```

Comment:
in a previous PR, @thomwolf suggested that if a kwarg (like `length_penalty`) is on the config, it should only be exposed through `generate_kwargs`).

Reply:
I would favor to (re-)define the kwargs like `max_length`, `min_length`, `do_sample`, length_penalty in this function because the defaults in the config are defaults for `generate()` in general, but there are not good defaults for summarization. And to me `pipelines` is about easy-to-use so I think good general defaults for summarization should be defined here. Happy to discuss.

Reply:
Ok that's quite an interesting case for T5 versus Bart because in Bart the pretrained model are task-specific so it make sense to have the task-specific generation hyper-parameters in the config while for T5 the pretrained model is generic so it would make more sense to have a way to specify task specific generation HP in the config.

In general, I'm not a fan of having model-specific hyper-parameters in general classes like pipelines.

What I would propose is maybe the following for T5 (open to discussion):
- have generic generation HP in the config like now
- also have a dict of class specific generation HP in the config which can override the generic HP.

Ex of config dic:
```python
{
max_length: 100,  # generic generation HP
length_penalty: 1.0,
task_specific_generation_params: {
     "summarization": {  # task id (e.g. name of the pipeline?)
          max_length: 140,
          length_penalty: 2.0
      },
     "translation_en_to_de": {
          max_length: 160,
          length_penalty: 3.0
      },
    },
}
```

What do you think?

Reply:
Yeah, I like that idea! 
Two questions:

- Should we enforce that all parameters defined in `task_specific_generation_params` have to be already defined so that they can only override? 
- Do we allow "summarization" and "translation" if the config has no `task_specific_generation_params` defined?